distributed/tests/test_s3.py

```diff
@@ -27,60 +27,154 @@
                                    b'{"amount": 700, "name": "Charlie"}\n'
                                    b'{"amount": 800, "name": "Dennis"}\n')}
 
-
-
-def test_get_list_of_summary_objects():
-    L = get_list_of_summary_objects(test_bucket_name, prefix='test/accounts',
-                                    anon=True)
+@pytest.yield_fixture
+def s3():
+    # could do with a bucket with write privileges.
+    yield S3FileSystem(anon=True)
+
+
+def test_s3_filesystem(s3):
+    fn = 'distributed-test/nested/file1'
+    assert fn in s3.walk('distributed-test')
+    assert s3.exists(fn)
+    assert not s3.exists(fn+'another')
+    data = b'hello\n'
+    assert s3.cat(fn) == data
+    assert s3.head(fn, 3) == data[:3]
+    assert s3.tail(fn, 3) == data[-3:]
+    assert s3.info(fn)['Size'] == len(data)
+    with pytest.raises(OSError):
+        s3.info(fn+'another')
+    assert s3.du(test_bucket_name, deep=True)[fn] == len(data)
+    assert fn not in s3.ls('distributed-test/')
+    assert fn in s3.ls('distributed-test/nested/')
+    assert fn in s3.ls('distributed-test/nested')
+    assert fn in s3.ls('s3://distributed-test/nested/')
+    assert fn not in s3.glob('distributed-test/')
+    assert fn not in s3.glob('distributed-test/*')
+    assert fn in s3.glob('distributed-test/nested')
+    assert fn in s3.glob('distributed-test/nested/*')
+    assert fn in s3.glob('distributed-test/nested/file*')
+    assert fn in s3.glob('distributed-test/*/*')
```

Comment:
These are great.  Thanks!

I recommend splitting some of these off into separate tests.  One for glob, one for ls, one for cat/head/tail, etc..  This helps us to quickly isolate what breaks when we make changes.